### PR TITLE
Colormap

### PR DESCRIPTION
**Pull request by sterlingbaldwin**

### CODE REVIEW

**Comment by sterlingbaldwin** [comment]
Not quite ready yet.

**Comment by downiec** [comment]
As part of your PR, could you include the Jupyter Lab extensions/packages that you mentioned would be good to install?

**Comment by sterlingbaldwin** [comment]
The "favorites" extension seams useful, but I dont see any difference with the other one. Also apparently the "open-from-path" didnt mean what I thought it did. I'll add the favorites for the time being.

**Comment by downiec** [comment]
Once your branch has the latest master merged in, it'll be easier for me to review the code. Is the code review the only thing left for this branch or is it still a draft?

File: src/components/ColormapEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+  DropdownMenu,
+  DropdownToggle,
+  Input,
+  InputGroup,
+  InputGroupAddon,
+  ListGroup,
+  ListGroupItem
+} from "reactstrap";
+
+const default_colormaps = ['AMIP',
```
Comment (downiec) [inline src/components/ColormapEditor.tsx]: This all looks good to me! But I think the default_colormaps const should be moved to the constants.ts file, no? That's where we have the default GM and other stuff.

File: src/CodeInjector.ts [hunk 2]
```diff
@@ -486,6 +486,24 @@ canvas = vcs.init(display_target='off')`;
...
 
+  /**
+   * Updates the colormap thats to be used by the selected graphics method
+   * @param gm_name: the name of the graphics method to update the color map for
+   * @param cm_name: the name of the color map
+   * @returns Promise<[number, string]> - A promise for when the cell code has executed containing
+   * the cell's index and output result
+   */
+  public async updateColormapName(gm_name: string, gm_group: string, cm_name: string): Promise<[number, string]>{
+    let cmd = `${gm_name}_${gm_group}.colormap = '${cm_name}'`;
```
Comment (downiec) [inline src/CodeInjector.ts]: There is some kind of issue going on with the graphic method and colormap code injection. I tried selecting a colormap but it gives an error due to the graphic method not being initialized. I was trying to look where the issue is but it doesn't seem straight forward, so let me know if you'd like me to troubleshoot it?

**Review by downiec — CHANGES_REQUESTED** [review]

File: src/CodeInjector.ts (hunk 2 above)
Comment (downiec) [reply src/CodeInjector.ts]: I think I found out the issue and will have a fix for it.

Comment (downiec) [reply src/CodeInjector.ts]: Minor styling issue: The variable names should match the camel case used by other variables in the file, not sure why this passed the linter

File: src/components/VCSMenu.tsx [hunk 3]
```diff
@@ -427,6 +452,7 @@ export default class VCSMenu extends React.Component<
         );
       } else {
         // Inject the plot
+        debugger;
```
Comment (downiec) [inline src/components/VCSMenu.tsx]: Should probably remove this.

**Review by downiec — APPROVED** [review]

**Pull request merged**